### PR TITLE
tty/pl011: basic TTY implementation for pl011.

### PR DESCRIPTION
**Pull request by pj1031999**
Basic TTY implementation for pl011.

### CODE REVIEW

File: sys/aarch64/pl011.c [hunk 1]
```diff
@@ -19,16 +21,33 @@
 #define UART0_BASE BCM2835_PERIPHERALS_BUS_TO_PHYS(BCM2835_UART0_BASE)
 #define UART_BUFSIZE 128
 
+static inline void set4(resource_t *r, int o, uint32_t v) {
+  bus_write_4(r, o, bus_read_4(r, o) | v);
+}
+
+static inline void clr4(resource_t *r, int o, uint32_t v) {
+  set4(r, o, bus_read_4(r, o) & ~v);
```
Comment (j-piecuch) [inline sys/aarch64/pl011.c:29]: Shouldn't this be `bus_write_4(r, o, bus_read_4(r, o) & ~v)`?

File: sys/aarch64/pl011.c [hunk 2]
```diff
@@ -57,103 +76,130 @@ static uint32_t pl011_getc(pl011_state_t *state) {
...
-      res = IF_FILTERED;
+  while (true) {
+    SCOPED_SPIN_LOCK(&pl011->lock);
+    pl011_try_bypass_txbuf(pl011, tty);
+    if (ringbuf_full(&pl011->tx_buf) || !ringbuf_getb(&tty->t_outq, &byte)) {
+      pl011_set_tty_outq_nonempty_flag(pl011, tty);
+      break;
     }
+    if (!ringbuf_empty(&pl011->tx_buf))
+      set4(pl011->regs, PL011COM_CR, PL011_CR_TXE);
```
Comment (j-piecuch) [inline sys/aarch64/pl011.c:152]: Is there a particular reason you moved these 2 lines here? In the `ns16550` version, they are inside the `if` above.

**Review by j-piecuch — CHANGES_REQUESTED** [review]

File: sys/aarch64/pl011.c (hunk 1 above)
Comment (pj1031999) [reply sys/aarch64/pl011.c]: It should. Fixed.

File: sys/aarch64/pl011.c (hunk 2 above)
Comment (pj1031999) [reply sys/aarch64/pl011.c]: It's my fault. Fixed.

**Review by j-piecuch — APPROVED** [review]

**Review by cahirwpz — APPROVED** [review]

**Pull request merged**